src/Shared/SecurityHelper/SecurityHelper.cs

```diff
@@ -22,6 +22,12 @@ internal static class SecurityHelper
         /// <param name="additionalPrincipal">The <see cref="ClaimsPrincipal"/> containing <see cref="ClaimsIdentity"/> to be added.</param>
         public static ClaimsPrincipal MergeUserPrincipal(ClaimsPrincipal? existingPrincipal, ClaimsPrincipal? additionalPrincipal)
         {
+            // For the first principal, just use the new principal rather than copying it
+            if (existingPrincipal == null && additionalPrincipal != null)
+            {
+                return additionalPrincipal;
+            }
```

Comment:
The method signature claims either could be null? Should we apply the same short circuit if they're reversed?
```suggestion
        public static ClaimsPrincipal? MergeUserPrincipal(ClaimsPrincipal? existingPrincipal, ClaimsPrincipal? additionalPrincipal)
        {
            // For the first principal, just use the new principal rather than copying it
            if (existingPrincipal == null || additionalPrincipal == null)
            {
                return existingPrincipal ?? additionalPrincipal;
            }
```

Reply:
Discussed offline and we are ok without this